src/main/java/com/scottlogic/sort/SortLikes.java

```diff
@@ -0,0 +1,28 @@
+package com.scottlogic.sort;
+
+import com.scottlogic.SortOrder;
+import com.scottlogic.SortStrategy;
+import com.scottlogic.UserPost;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class SortLikes implements SortStrategy {
+
+    @Override
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder sortOrder) {
+        if (inputList == null) {
+            return Arrays.asList();
+        }
+        if (sortOrder.equals(SortOrder.ASC)) {
+            Collections.sort(inputList, Comparator.comparing(UserPost::getLikeCount));
+        }
+        else {
+            Collections.sort(inputList, Comparator.comparing(UserPost::getLikeCount).reversed());
```

Comment:
Here's a weakness of composition: you're doing if (ASC) else ... a lot.
I wonder if you could share this behaviour across your strategies, like having a default implementation method on the interface that takes in a method (eg UserPost::getLikeCount) then you can reuse this code and swap out the comparator. Just a thought, I think we should move on now (but could revisit if you finish the tasks early)

Reply:
Yea would always like to revisit/talk about design patterns as early as possible so it's not foreign later - shall I leave it as it is or change the implementation?

Reply:
For now let's focus on java and testing, since you will have much more design patterns coming from the study group.
I'm happy for you to leave it as it is now as I haven't thought about a cleaner way to suggest.